app/models/assignment.rb

```diff
@@ -769,7 +769,7 @@ def section_groupings(section)
     groupings.select do |grouping|
       grouping.inviter.present? &&
       grouping.inviter.has_section? &&
-      grouping.inviter.section.id == section.id
+      grouping.section.id == section.id
```

Comment:
Good find---actually, with the new association, this entire block can be simplified. We no longer need to have the first two checks, and can instead use the [safe navigation operator `?.`](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/). Please read about that and then use this to simplify this code.